### ISSUE TITLE
llvm 12 regression: aarch64 f16 @reduce  with min

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#8130]
[upstream bug report](https://bugs.llvm.org/show_bug.cgi?id=49401)

This is causing behavior tests to fail on aarch64 with LLVM 12.

**Comment by mikdusan** [comment]
- reduction
```sh
zig build-obj reduction.zig -fno-emit-bin -femit-llvm-ir -target aarch64-linux-none
llc reduction.ll
```
```
LLVM ERROR: Cannot select: intrinsic %llvm.aarch64.neon.fminnmv
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
```

- `reduction.zig`
```zig
const std = @import("std");

export fn entry() void {
    var value: std.meta.Vector(4, f16) = [_]f16{ 1, -1, 1, -1 };
    var r1 = @reduce(.Min, value);
    var r2 = @reduce(.Max, value);
}

pub fn panic(msg: []const u8, error_return_trace: ?*@import("builtin").StackTrace) noreturn {
    while (true) {}
}
```

- stripped `reduction.ll`
```llvm
; ModuleID = 'reduction.ll'
source_filename = "reduction"
target datalayout = "e-m:e-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128"
target triple = "aarch64-unknown-linux-unknown"

%0 = type { i8*, i64 }
%1 = type { i64, %2 }
%2 = type { i64*, i64 }

@0 = internal unnamed_addr constant void (%0*, %1*)* @2, align 8
@1 = internal unnamed_addr constant i2 -2, align 1

; Function Attrs: nobuiltin noreturn nounwind
define internal fastcc void @2(%0* nonnull readonly align 8 %0, %1* align 8 %1) unnamed_addr #0 {
  %3 = alloca %1*, align 8
  store %1* %1, %1** %3, align 8
  br label %4

4:                                                ; preds = %4, %2
  br label %4
}

; Function Attrs: nofree nosync nounwind readnone speculatable willreturn
declare void @llvm.dbg.declare(metadata, metadata, metadata) #1

; Function Attrs: nobuiltin nounwind
define void @entry() #2 {
  %1 = alloca <4 x half>, align 8
  %2 = alloca half, align 2
  %3 = alloca half, align 2
  store <4 x half> <half 0xH3C00, half 0xHBC00, half 0xH3C00, half 0xHBC00>, <4 x half>* %1, align 8
  %4 = load <4 x half>, <4 x half>* %1, align 8
  %5 = call half @llvm.vector.reduce.fmin.v4f16(<4 x half> %4)
  store half %5, half* %2, align 2
  %6 = load <4 x half>, <4 x half>* %1, align 8
  %7 = call half @llvm.vector.reduce.fmax.v4f16(<4 x half> %6)
  store half %7, half* %3, align 2
  ret void
}

; Function Attrs: nofree nosync nounwind readnone willreturn
declare half @llvm.vector.reduce.fmin.v4f16(<4 x half>) #3

; Function Attrs: nofree nosync nounwind readnone willreturn
declare half @llvm.vector.reduce.fmax.v4f16(<4 x half>) #3

attributes #0 = { nobuiltin noreturn nounwind "frame-pointer"="all" }
attributes #1 = { nofree nosync nounwind readnone speculatable willreturn }
attributes #2 = { nobuiltin nounwind "frame-pointer"="all" }
attributes #3 = { nofree nosync nounwind readnone willreturn }

!llvm.module.flags = !{!0, !1}

!0 = !{i32 2, !"Debug Info Version", i32 3}
!1 = !{i32 2, !"Dwarf Version", i32 4}
```

**Comment by andrewrk** [comment]
Thanks!

**Comment by LemonBoy** [comment]
[LLVM patch](https://reviews.llvm.org/D97840)

**Comment by mikdusan** [comment]
fixed in 12.0.0-rc3 (and 12.0.0-rc4)
